setupext.py

```diff
@@ -531,6 +540,38 @@ def do_custom_build(self):
         """
         pass
 
+    def install_help_msg(self):
+        """
+        The help message to show if the package is not installed. The help
+        message shown depends on whether some class variables are present.
+        """
+        def _try_managers(*managers):
+            for manager in managers:
+                pkg_name = self.pkg_names.get(manager, None)
+                if pkg_name:
+                    try:
+                        _ = check_output(["which", manager],
```

Comment:
There is a [`shutil.which`](https://docs.python.org/3/library/shutil.html), but unfortunately, it's 3.3+. It's probably not worth it to conditionally call it, but can you add a comment about switching to it when we drop 2.7?